### ISSUE TITLE
Model.register is failing when I pass a ParameterString for model_package_group_name

### ISSUE THREAD

**Issue by rawwar** [aws/sagemaker-python-sdk#4139]
**Describe the bug**
I am trying to create a SageMaker Pipeline. I have Training step where I am using a tensorflow estimator to train a model. I am trying to register the trained model using ModelStep. But, when passing `model_package_group_name`(parameterString) to model.register, its failing.

**To reproduce**

My parameterString variable declaration:
```
model_registry_package = ParameterString(name="ModelGroup", default_value="default-registry")
```

My Estimator and train step

```
from sagemaker.tensorflow import TensorFlow
estimator = TensorFlow(entry_point='train.py',
                       role=role,
                       instance_count=1,
                       framework_version='2.10',
                       instance_type="ml.g5.xlarge",
                       dependencies=['/home/lib/']
                       output_path = Join(on="/",values=['s3:/', bucket_name, model_path_prefix]),
                       py_version='py39',
                       use_spot_instances=False,
                       max_run=5000,
                       model_dir = False,
                       module_dir = False,
                       SM_MODULE_DIR = False
                      )

step_train = TrainingStep(name='ModelTraining',
                          estimator=estimator,
                          inputs={
                                "train": TrainingInput(
                                    s3_data=step_preprocess_data.properties.ProcessingOutputConfig.Outputs[
                                        "train"
                                    ].S3Output.S3Uri,
                                    content_type="text/csv",
                                ),
                                "validation": TrainingInput(
                                    s3_data=step_preprocess_data.properties.ProcessingOutputConfig.Outputs[
                                        "validation"
                                    ].S3Output.S3Uri,
                                    content_type="text/csv",
                                ),
                               "test": TrainingInput(
                                    s3_data=step_preprocess_data.properties.ProcessingOutputConfig.Outputs[
                                        "test"
                                    ].S3Output.S3Uri,
                                    content_type="text/csv",
                                ),
                            })
```
Model Register step:

```
from sagemaker.model import Model
from sagemaker.workflow.model_step import ModelStep

model = Model(
    image_uri=estimator.training_image_uri(),
    model_data=step_train.properties.ModelArtifacts.S3ModelArtifacts,
    role=role,
    sagemaker_session=session
)
step_model_create = ModelStep(
   name="MyModelCreationStep",
   step_args=model.register(
       model_package_group_name=model_registry_package,
       content_types=["application/json"],
       response_types=["application/json"]),
)
```

**Expected behavior**
I should be able to register the model by passing model package name as a parameterString 

**Screenshots or logs**
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[36], line 1
----> 1 model.register(
      2        model_package_group_name=model_registry_package,
      3        content_types=["application/json"],
      4        response_types=["application/json"])

File /opt/conda/lib/python3.10/site-packages/sagemaker/workflow/pipeline_context.py:311, in runnable_by_pipeline.<locals>.wrapper(*args, **kwargs)
    307         return context
    309     return _StepArguments(retrieve_caller_name(self_instance), run_func, *args, **kwargs)
--> 311 return run_func(*args, **kwargs)

File /opt/conda/lib/python3.10/site-packages/sagemaker/model.py:505, in Model.register(self, content_types, response_types, inference_instances, transform_instances, model_package_name, model_package_group_name, image_uri, model_metrics, metadata_properties, marketplace_cert, approval_status, description, drift_check_baselines, customer_metadata_properties, validation_specification, domain, task, sample_payload_url, framework, framework_version, nearest_model_name, data_input_configuration, skip_model_validation)
    479     container_def = {
    480         "Image": self.image_uri,
    481         "ModelDataUrl": self.model_data,
    482     }
    484 model_pkg_args = sagemaker.get_model_package_args(
    485     content_types,
    486     response_types,
   (...)
    503     skip_model_validation=skip_model_validation,
    504 )
--> 505 model_package = self.sagemaker_session.create_model_package_from_containers(
    506     **model_pkg_args
    507 )
    508 if isinstance(self.sagemaker_session, PipelineSession):
    509     return None

File /opt/conda/lib/python3.10/site-packages/sagemaker/session.py:3772, in Session.create_model_package_from_containers(self, containers, content_types, response_types, inference_instances, transform_instances, model_package_name, model_package_group_name, model_metrics, metadata_properties, marketplace_cert, approval_status, description, drift_check_baselines, customer_metadata_properties, validation_specification, domain, sample_payload_url, task, skip_model_validation)
   3765         _create_resource(
   3766             lambda: self.sagemaker_client.create_model_package_group(
   3767                 ModelPackageGroupName=request["ModelPackageGroupName"]
   3768             )
   3769         )
   3770     return self.sagemaker_client.create_model_package(**request)
-> 3772 return self._intercept_create_request(
   3773     model_pkg_request, submit, self.create_model_package_from_containers.__name__
   3774 )

File /opt/conda/lib/python3.10/site-packages/sagemaker/session.py:5513, in Session._intercept_create_request(self, request, create, func_name)
   5496 def _intercept_create_request(
   5497     self,
   5498     request: typing.Dict,
   (...)
   5501     # pylint: disable=unused-argument
   5502 ):
   5503     """This function intercepts the create job request.
   5504 
   5505     PipelineSession inherits this Session class and will override
   (...)
   5511         func_name (str): the name of the function needed intercepting
   5512     """
-> 5513     return create(request)

File /opt/conda/lib/python3.10/site-packages/sagemaker/session.py:3762, in Session.create_model_package_from_containers.<locals>.submit(request)
   3761 def submit(request):
-> 3762     if model_package_group_name is not None and not model_package_group_name.startswith(
   3763         "arn:"
   3764     ):
   3765         _create_resource(
   3766             lambda: self.sagemaker_client.create_model_package_group(
   3767                 ModelPackageGroupName=request["ModelPackageGroupName"]
   3768             )
   3769         )
   3770     return self.sagemaker_client.create_model_package(**request)

AttributeError: 'ParameterString' object has no attribute **'startswith'**
```

**System information**
A description of your system. Please provide:
- **SageMaker Python SDK version**: 2.187.0
- **Framework name (eg. PyTorch) or algorithm (eg. KMeans)**: Tensorflow
- **Framework version**: 2.10
- **Python version**:  3.10.6
- **CPU or GPU**: CPU
- **Custom Docker image (Y/N)**: No.  

**Additional context**
I am running this in Sagemaker Studio with Data Science 3.0 image and python3 kernel

**Comment by francisco-parrilla-lner** [comment]
I know that the documentation states that the `Model.register()` method accepts a `PipelineVariable` for the parameter `model_package_group_name`.

I believe a quick fix would to do `model_registry_package= model_registry_package.default_value` instead.

**Comment by qidewenwhen** [comment]
Adding a note: 
It's this change newly added in this year led to this regression:

https://github.com/aws/sagemaker-python-sdk/blame/d355d5b01b892c7baa314c1ce0c1243b850653b9/src/sagemaker/session.py#L3867

Next steps:
- we should firstly add one more condition to check if it's a PipelineVariable first, e.g.
```
            if is_pipeline_variable(model_package_group_name):
                   logger.warn("The runtime value of model_package_group_name should be valid......")
            elif model_package_group_name is not None and not model_package_group_name.startswith(
                "arn:"
            ):
                _create_resource(
                    lambda: self.sagemaker_client.create_model_package_group(
                        ModelPackageGroupName=request["ModelPackageGroupName"]
                    )
                )
```

Tracking the fix in backlog

**Comment by qidewenwhen** [comment]
Hi @rawwar, sorry for the delay.

I'm not able to reproduce the issue with the code sample provided. 
After a second look, I figured out the issue may be caused because of **not using `PipelineSession` while defining the model**:

```
model = Model(
    image_uri=estimator.training_image_uri(),
    model_data=step_train.properties.ModelArtifacts.S3ModelArtifacts,
    role=role,
    sagemaker_session=session # <<<<<<<<<<<<<<<<<< should be PipelineSession object
)
```

The conclusion is from this log trace:

```
File /opt/conda/lib/python3.10/site-packages/sagemaker/workflow/pipeline_context.py:311, in runnable_by_pipeline.<locals>.wrapper(*args, **kwargs)
    307         return context
    309     return _StepArguments(retrieve_caller_name(self_instance), run_func, *args, **kwargs)
--> 311 return run_func(*args, **kwargs)
```
If `PipelineSession` is used (the expected usage), it should not reach line 311, rather, it should hit the line 309 (as per the [code](https://github.com/aws/sagemaker-python-sdk/blob/08c8a3a1c3e583385cf34e08006c9600b68d3bd7/src/sagemaker/workflow/pipeline_context.py#L310)), which accordingly would not invoke the `submit` function to call the create_model_package API directly in pipeline compile time (i.e when generating the pipeline definition in Python SDK).

```
   3761 def submit(request):
-> 3762     if model_package_group_name is not None and not model_package_group_name.startswith(
   3763         "arn:"
   3764     ):
   3765         _create_resource(
   3766             lambda: self.sagemaker_client.create_model_package_group(
   3767                 ModelPackageGroupName=request["ModelPackageGroupName"]
   3768             )
   3769         )
   3770     return self.sagemaker_client.create_model_package(**request)

```

To sum up, please update the `sagemaker_session` used to define the Model to be a PipelineSession object, which should resolve the issue.
In addition, please find more info w.r.t PipelineSession in https://sagemaker.readthedocs.io/en/stable/amazon_sagemaker_model_building_pipeline.html#pipeline-session

**Comment by qidewenwhen** [comment]
Closing the issue as we did not hear back in past 3 weeks. Feel free to reopen if you have further questions. Thanks!
